cpp/src/groupby/groupby_without_aggregation.cu

```diff
@@ -91,11 +91,22 @@ std::pair<cudf::table, gdf_column> gdf_group_by_without_aggregations(
 
   gdf_size_type nrows = input_table.num_rows();
 
+  // Ask if input table has nulls
+  auto get_key_table = [&]() {
+    std::vector<gdf_column *> key_columns;
```

Comment:
```suggestion
    std::vector<gdf_column *> key_columns(num_key_cols);
    std::transform(key_columns.begin(), key_columns.end(), std::begin(key_col_indices), 
                             [&](auto i) { return input_table.get_column(key_col_indices[i]); });
```

Reply:
I changed the lambda function to a simple one raw loop. The tests were updated previously in order to match more accurately the cases which this fix is covering.

Reply:
Please replace the raw loop with a std::transform as I suggested.  Suggested viewing. https://channel9.msdn.com/Events/GoingNative/2013/Cpp-Seasoning